client/ayon_core/pipeline/load/plugins.py

```diff
@@ -251,15 +252,50 @@ class ProductLoaderPlugin(LoaderPlugin):
     """
 
 
+class PrePostLoaderHookPlugin:
+    """Plugin that should be run before or post specific Loader in 'loaders'
+
+    Should be used as non-invasive method to enrich core loading process.
+    Any studio might want to modify loaded data before or after
+    they are loaded without need to override existing core plugins.
+    """
+    @classmethod
+    def is_compatible(cls, Loader: LoaderPlugin) -> bool:
+        pass
+
+    def pre_load(
+        self,
+        context: dict,
+        name: str | None = None,
+        namespace: str | None = None,
+        options: dict | None = None,
+    ):
+        pass
+
+    def post_load(
+        self,
+        context: dict,
+        name: str | None = None,
+        namespace: str | None = None,
+        options: dict | None = None,
+        container: dict | None = None,  # (ayon:container-3.0)
+    ):
+        pass
+
+    def switch(self, container, context):
```

Comment:
There probably should be `pre_switch`, `post_switch`, `pre_update`, `post_update`.

Reply:
question is then if it should chain these in a case where `switch()` calls directly `update()` (which is the case probably most of the time). It probably should.